solidity/contracts/peripherals/Auction.sol

```diff
@@ -0,0 +1,49 @@
+// SPDX-License-Identifier: UNICENSE
+pragma solidity 0.8.30;
+uint256 constant AUCTION_TIME = 1 weeks;
+
+contract Auction {
+	mapping(address => uint256) public purchasedRep;
+	uint256 public totalRepPurchased;
+	uint256 public repAvailable;
+	uint256 public auctionStarted;
+	uint256 public ethAmountToBuy;
+	bool public finalized;
+	address immutable owner;
+
+	event Participated(address user, uint256 repAmount, uint256 ethAmount, uint256 totalRepPurchased);
+	event FinalizedAuction(address user, uint256 repAmount, uint256 ethAmount);
+	event AuctionStarted(uint256 ethAmountToBuy, uint256 repAvailable);
+
+	constructor(address _owner) {
+		owner = _owner;
+	}
+
+	function participate(uint256 repToBuy) public payable {
+		require(auctionStarted > 0, 'Auction needs to have started');
+		require(!finalized, 'Already finalized');
+		require(msg.value > 0, 'need to invest with eth!');
+		require(address(this).balance <= ethAmountToBuy, 'attempting to overfund');
+		require(totalRepPurchased + repToBuy <= repAvailable, 'attempt to buy too much rep');
+		purchasedRep[msg.sender] = repToBuy; // todo, currently anyone can buy with any price
```

Comment:
For prototyping, how hard would it be to just have a linear countdown since start and just require the price matches exactly?